tests/accessor/local_accessor_access_among_work_items.h

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides tests for the properties of sycl::local_accessor that can access to
+//  the memory shared among work-items
+//
+*******************************************************************************/
+#ifndef SYCL_CTS_LOCAL_ACCESSOR_ACCESS_AMONG_WORK_ITEMS_H
+#define SYCL_CTS_LOCAL_ACCESSOR_ACCESS_AMONG_WORK_ITEMS_H
+#include "accessor_common.h"
+
+namespace local_accessor_access_among_work_items {
+using namespace sycl_cts;
+using namespace accessor_tests_common;
+
+/**
+ * @brief Provides a functor that provides verification that local_accessor can
+ *        access to the memory shared among work-items
+ * @tparam T Current data type
+ * @tparam DimensionTypeT Current current dimension size
+ */
+template <typename T, typename DimensionTypeT>
+class run_test {
+  static constexpr int Dimension = DimensionT::value;
+
+ public:
+  /**
+   * @brief Functor that provides verification that local_accessor can access to
+   *        the memory shared among work-items
+   * @param type_name Current data type string representation
+   */
+  void operator()(const std::string& type_name) {
+    auto queue = util::get_cts_object::queue();
+
+    auto section_name = get_section_name<Dimension>(
+        type_name,
+        "Verify possibility to access to the memory shared among work-items. "
+        "[local_accessor]");
+    SECTION(section_name) {
+      T values_arr[2] = {1, 2};
+      const sycl::range range(2);
+      constexpr T new_value = 5;
+      constexpr T new_assign_val = 20;
+      bool acc_val_is_equal_to_expected = false;
+      {
+        sycl::buffer<bool> val_is_equal_to_expected_buffer(
+            &acc_val_is_equal_to_expected, sycl::range(1));
+        queue.submit([&](sycl::handler& cgh) {
+          auto val_is_equal_to_expected_acc =
+              val_is_equal_to_expected_buffer
+                  .template get_access<sycl::access_mode::write>(cgh);
+          cgh.parallel_for(
+              sycl::nd_range(range, range), [=](sycl::nd_item item) {
+                sycl::local_accessor<T, Dimension> acc(range, cgh);
+                T& acc_elem = array_acc[0];
+
+                if (item.get_global_id(0) != 0) {
```

Comment:
Looks like you try to set value in one work item, synchronize work items with `group_barrier` and check that value have been set correctly from another work item. If you leave a comment to this algorithm it would be nice.

Reply:
Comment has been added in this commit: https://github.com/KhronosGroup/SYCL-CTS/pull/364/commits/5db87bf32e0364c53ccc4bcf6341571d4c8c3a19